recipes/gtsam/all/conanfile.py

```diff
@@ -0,0 +1,166 @@
+            self._cmake.definitions["GTSAM_BUILD_WRAP"] = self.options.build_wrap
+            self._cmake.definitions["GTSAM_WRAP_SERIALIZATION"] = self.options.wrap_serialization
+            self._cmake.definitions["GTSAM_INSTALL_MATLAB_TOOLBOX"] = self.options.install_matlab_toolbox
+            self._cmake.definitions["GTSAM_INSTALL_CYTHON_TOOLBOX"] = self.options.install_cython_toolbox
+            self._cmake.definitions["GTSAM_INSTALL_CPPUNITLITE"] = self.options.install_cppunitlite
+            self._cmake.definitions["GTSAM_INSTALL_GEOGRAPHICLIB"] = False #Use conan-provided geographiclib
+            self._cmake.definitions["GTSAM_USE_SYSTEM_EIGEN"] = True #Set to false to use eigen sources contained in GTSAM
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self): #Needed to build GTSam as a subproject using the conan CMake Wrapper.
+        for cmake in (os.path.join(self._source_subfolder, "gtsam", "CMakeLists.txt"),
+                      os.path.join(self._source_subfolder, "wrap", "CMakeLists.txt"),
+                      os.path.join(self._source_subfolder, "CMakeLists.txt"),
+                      os.path.join(self._source_subfolder, "cmake", "GtsamPythonWrap.cmake")):
+            tools.replace_in_file(cmake,
+                                  "${CMAKE_SOURCE_DIR}",
+                                  "${GTSAM_SOURCE_DIR}")
+        tools.replace_in_file(os.path.join(self._source_subfolder, "gtsam", "CMakeLists.txt"),
+                              "${CMAKE_BINARY_DIR}",
+                              "${GTSAM_BINARY_DIR}")
+        if self.settings.os == "Windows": #compiler.runtime field only exists on Windows
+            tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "GtsamBuildTypes.cmake"),
+                                  "/MD ",
+                                  "/{} ".format(self.settings.compiler.runtime))
+            tools.replace_in_file(os.path.join(self._source_subfolder, "cmake", "GtsamBuildTypes.cmake"),
+                                  "/MDd ",
+                                  "/{} ".format(self.settings.compiler.runtime))
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.16.2")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
+                raise ConanInvalidConfiguration ("GTSAM requires MSVC >= 15")
+        if self.settings.os == "Linux":
+            if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < 5:
+                raise ConanInvalidConfiguration ("This recipe won't build on gcc 4.9 until uilianries:hotfix/gcc49-glibc gets merged")
```

Comment:
This recipe didn't build a few months ago because of some error with gcc 4.9, that's why we added this. See [CCI log](https://c3i.jfrog.io/c3i/misc/logs/pr/684/12/gtsam/4.0.2/bc9232061b17482ce34ba48f70fb6e409d52e78f/create_stdout.txt)
Now it is failing again, with a similar error but on gcc 5. See [CCI log](https://c3i.jfrog.io/c3i/misc/logs/pr/684/21/gtsam/4.0.2/aaf2e47ba675f9f7caa0c47e98c0ecf0a8270296/create_stdout.txt)
There was a PR made by @uilianries to fix this, but it never got merged.
Did we choose a different approach for this?

Reply:
@ericriff , please, remove cmake from build requirements. Now CCI provides 3.16.4 https://github.com/conan-io/conan-center-index/wiki/Deployment-Changelog

About CMake package, we don't a have solution yet. It's affecting not only cmake, but also other installer packages.